### PR TITLE
recovery from channel failure

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Add a test case that shows that currently recovery from channel failures isn't working.

If you run it, you will see:

```
2023/03/06 13:42:55 gorabbit WARN: successfully reconnected to amqp server
2023/03/06 13:42:55 gorabbit INFO: successful recovery from: Exception (405) Reason: "RESOURCE_LOCKED - cannot obtain exclusive access to locked queue 'test' in vhost '/'. It could be originally declared on another connection or the exclusive property value does not match that of the original declaration."
```

But there is actually not connection established to the server.

Related to https://github.com/wagslane/go-rabbitmq/issues/99

### CODE REVIEW

**Comment by johanneswuerbach** [comment]
@wagslane rebased this branch, but also with the latest changes in main, the reconnect test fails.

File: integration/main_test.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+	require.Len(receivedMessages, 2)
+
+	var receivedMsg TestMessage
+
+	require.NoError(json.Unmarshal(receivedMessages[0].Body, &receivedMsg))
+	require.Equal(msg, receivedMsg)
+}
+
+func TestBasicReconnectBrokenChannel(t *testing.T) {
+	t.Skip("This test fails currently, but shouldn't")
```
Comment (johanneswuerbach) [inline integration/main_test.go:161]: Marked this as skipped for now, but from my understand the connection should eventually recover here.

**Comment by wagslane** [comment]
@johanneswuerbach Thanks for doing this! If you remove the "skip" line so that it breaks as expected I'll merge this in and I or someone else can work on a fix when we have time

**Comment by johanneswuerbach** [comment]
Done :-)

**Pull request merged**